sympy/matrices/dense.py

```diff
@@ -35,15 +36,16 @@ def _compare_sequence(a, b):
     # tuple
     return tuple(a) == tuple(b)
 
+@sympify_method_args
 class DenseMatrix(MatrixBase):
 
     is_MatrixExpr = False  # type: bool
 
     _op_priority = 10.01
     _class_priority = 4
 
+    @sympify_return([('other', 'DenseMatrix')], NotImplemented)
```

Comment:
Can't we add this to SparseMatrix such that it can be symmetric change?

Reply:
okay , would do that.

Reply:
The decorator can be  used when sympifying code, since the SparseMatrix isn't sympifying , it shouldn't be added in SparseMatrix

Reply:
I'd rather change that SparseMatrix to sympify the arguments before calls
It is problematic if operations supported in DenseMatrix is not supported by SparseMatrix only because of sympify
For example,
```
from sympy import *
import mpmath

m = mpmath.matrix([[1, 2], [3, 4]])
Matrix([[1, 2], [3, 4]]) == m # True
SparseMatrix([[1, 2], [3, 4]]) == m # False